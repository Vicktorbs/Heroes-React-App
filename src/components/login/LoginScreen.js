import React, { useContext } from 'react'
import { AuthContext } from '../../auth/AuthContext'
import { types } from '../../types/types'
// import { authReducer } from '../../auth/autReducer'

export const LoginScreen = ({ history }) => {
    
    const { dispatch } = useContext(AuthContext);

    const handleLogin = () => {
        // history.push('/');
        dispatch({
            type : types.login, 
            payload: {
                name: 'Victor'
            }
        });
        history.replace('/');
    }

    return (
        <div className="container mt-4">
            <h3>Login!</h3>
            <hr />
            <button 
                className="btn btn-success"
                onClick={ handleLogin }
            >
                Login
            </button>
        </div>
    )
}
