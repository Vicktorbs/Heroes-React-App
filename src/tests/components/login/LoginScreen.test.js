import React from 'react';
import { mount } from "enzyme";
import { LoginScreen } from '../../../components/login/LoginScreen';
import { AuthContext } from '../../../auth/AuthContext';
import { MemoryRouter } from 'react-router-dom';
import { types } from '../../../types/types';

describe('Test on <LoginScreen /> component', () => {
    
    const contexValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }
    
    const wrapper = mount(
        <AuthContext.Provider value={ contexValue }>
            <MemoryRouter>
                <LoginScreen history={ historyMock } />
            </MemoryRouter>
        </AuthContext.Provider>
    );

    test('should show itself correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should use dispatch and navigate', () => {
        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();
        expect(contexValue.dispatch).toHaveBeenCalledWith({
            type : types.login, 
            payload: {
                name: 'Victor'
            }
        });
        expect(historyMock.replace).toHaveBeenCalled();
        expect(historyMock.replace).toHaveBeenCalledWith('/');

        localStorage.setItem('lastLocation', '/dc');
        handleClick();

        expect(historyMock.replace).toHaveBeenCalledWith('/dc');
    })
    
})
