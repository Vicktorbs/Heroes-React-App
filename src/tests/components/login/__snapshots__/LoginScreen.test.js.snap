// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <LoginScreen /> component should show itself correctly 1`] = `
<div
  className="container mt-4"
>
  <h3>
    Login!
  </h3>
  <hr />
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
